Assets/Scripts/Level01Controller.cs

```diff
@@ -113,45 +47,60 @@ public class Level01Controller : MonoBehaviour,LevelController {
 			}
 		}
 	}
-	private void LoadGameControlData(Level01Data leveldata){
-		Debug.Log ("Loading Game (JSON):"+JsonUtility.ToJson(leveldata));
-		LoadSavedPlayer (leveldata.playerdata);
-		LoadSavedSheepData (leveldata.sheepdata);
-		LoadSavedBuildingData (leveldata.buildingdata);
+	public void LoadGameControlData(Level01Data leveldata){
+		Debug.Log ("Loading Game (JSON):" + JsonUtility.ToJson (leveldata));
+		if (leveldata != null) {
+			LoadSavedPlayer (leveldata.playerdata);
+			LoadListSheepData (leveldata.sheepdata);
+			LoadListBuildingData (leveldata.buildingdata);
+
+			scoreManager.setPlayerScore (leveldata.score);
+		}
+	}
 
-		scoreManager.setPlayerScore(leveldata.score);
+	private List<SheepData> BuildListSheepData(){
+		List<SheepData> sheepDataList = new List<SheepData>();
+		foreach(Transform sheep in sheepOBJHolder.transform){
+			SheepData sd = sheep.gameObject.GetComponent<SheepEnemy> ().BuildSheepData ();
+			sheepDataList.Add (sd);
+		}
+		return sheepDataList;
+	}
+
+	private List<BuildingData> BuildListBuildingData(){
```

Comment:
For example, lines 61 - 68 are nearly identical to 70 - 78. I would think this functionality could be abstracted out into a single function at least, if not into a util function in the LevelController abstract class

Reply:
I think we would need to flush out more differences between the Buildings,Sheep,and PlayerData classes to really see if this should be separated or consolidated into one BuildChildrenOfObject() method. For Example, PlayerData is ultra basic, just really houses a transformData. BuildingData adds a Health field, and SheepData could really expand to include the RageMeter, the WanderTarget, etc. I just don't think we have gotten the chance to expand some of the differences until we tackle more of the tasks down the line.